kong/resty/dns/client.lua

```diff
@@ -839,7 +813,37 @@ local function syncQuery(qname, r_opts, try_list)
   local key = qname..":"..r_opts.qtype
   local item = queue[key]
 
-  -- if nothing is in progress, we start a new async query
+  local is_yieldable = in_yieldable_phase()
+
+  -- If nothing is in progress and in the yieldable phase,
+  -- we start a new sync query
+
+  if not item and is_yieldable then
```

Comment:
So if we want to query DNS in the `init_worker` phase, the logic can not query any DNS name, Is this a logic bug for Kong,. In our old logic, we can trigger DNS query in any openresty phase, I think we need to think about this carefully.

Reply:
yeah, if it is in init_worker phase, it will run the following logic which is kept by this pr. See init_worker phase's running logic: https://github.com/Kong/kong/pull/11900/files#diff-2001be5a8987462f9edc79d36a2f3c3a7138ae56a2aca1ff75573d19127edf7eR847( also for other unyieldable phases)